e2e-tests/playwright/e2e/plugins/adoption-Insights/adoption-Insights.spec.ts

```diff
@@ -0,0 +1,238 @@
+        });
+        const entries = panel.locator("tbody").locator("tr");
+        expect(await entries.count()).toBeGreaterThan(0);
+      });
+
+      test("Visited techdoc shows up in top techdocs", async () => {
+        const panel = page.locator(".v5-MuiPaper-root", {
+          hasText: "Top 3 techdocs",
+        });
+        const entries = panel.locator("tbody").locator("tr");
+        expect(await entries.count()).toBeGreaterThan(0);
+      });
+
+      test("Visited templates shows in top templates", async () => {
+        const panel = page.locator(".v5-MuiPaper-root", {
+          hasText: "Top 3 templates",
+        });
+        const entries = panel.locator("tbody").locator("tr");
+        expect(await entries.count()).toBeGreaterThan(0);
+      });
+
+      test("Changes are Reflecting in panels", async () => {
+        const titles = [
+          "catalog entities",
+          "techdocs",
+        ];
+
+        interface PanelState {
+          firstRow?: any;
+          initialViewsCount?: number;
+        }
+      
+        let state: Record<string, PanelState> = {
+          "catalog entities": {},
+          "techdocs": {},
+        };
+
+        for (const title of titles) {
+          const panel = page.locator(".v5-MuiPaper-root", { hasText: title }).last();
+          if (title === "catalog entities" || title === "techdocs" || title === "templates") {
+            state[title].firstRow = await testHelper.getVisibleFirstRowText(panel);
+      
+            if (title === "catalog entities") catalogEntitiesFirstEntry = state[title].firstRow;
+            else if (title === "techdocs") techdocsFirstEntry = state[title].firstRow;
+          }
+          const firstRow = panel.locator('table.v5-MuiTable-root tbody tr').first();
+          const firstEntry = firstRow.locator('td').first();
+          state[title].firstRow = firstRow;
+
+          let headerTxt: string;
+
+          if (title === "techdocs") {
+            headerTxt = techdocsFirstEntry[0];
+            state[title].initialViewsCount = Number(techdocsFirstEntry[1]); 
+            await testHelper.clickAndVerifyText(firstEntry, headerTxt);
+
+          } else if (title === "catalog entities") {
+            headerTxt = catalogEntitiesFirstEntry[0];
+            state[title].initialViewsCount = Number(catalogEntitiesFirstEntry[1]);
+            await testHelper.clickAndVerifyText(firstEntry, headerTxt);
+          }
+        }
+
+        for (const title of titles) {
+          await page.reload();
+          await testHelper.waitUntilApiCallSucceeds(page, '/api/adoption-insights/events');
+          await uiHelper.openSidebarButton("Administration");
+          await uiHelper.clickLink('Adoption Insights');
+          await testHelper.clickByText('Last 28 days'); 
+          await testHelper.selectOption("Today");
+          await testHelper.waitForPanelApiCalls(page);
+          await testHelper.waitUntilApiCallSucceeds(page, '/api/adoption-insights/events');
```

Comment:
can't this go before the loop? it's not like the page is changing between iterations